lua/bufferline/render.lua

```diff
@@ -757,125 +523,112 @@ local function generate_tabline(bufnrs, refocus)
   local current_buffer_index = nil
   local current_buffer_position = 0
 
+  local inactive_separator = options.icons().inactive.separator
+
   for i, bufnr in ipairs(bufnrs) do
+    local activity = Buffer.activities[Buffer.get_activity(bufnr)]
+
     local buffer_data = state.get_buffer_data(bufnr)
+    local buffer_hl = hl_tabline('Buffer' .. activity .. (
+      buf_get_option(bufnr, 'modified') and 'Mod' or ''
+    ))
     local buffer_name = buffer_data.name or '[no name]'
 
     buffer_data.real_width    = Layout.calculate_width(layout.base_widths[i], layout.padding_width)
     buffer_data.real_position = current_buffer_position
 
-    local activity = Buffer.get_activity(bufnr)
-    local is_inactive = activity == 1
-    local is_current = activity == 4
-    local is_modified = buf_get_option(bufnr, 'modified')
-    -- local is_closing = buffer_data.closing
-    local is_pinned = state.is_pinned(bufnr)
+    local icons_option = state.icons(bufnr, activity)
 
-    local status = HL_BY_ACTIVITY[activity]
-    local mod = is_modified and 'Mod' or ''
+    --- Prefix this value to allow an element to be clicked
+    local clickable = click_enabled and ('%' .. bufnr .. '@BufferlineMainClickHandler@') or ''
 
-    local separatorPrefix = hl_tabline('Buffer' .. status .. 'Sign')
-    local separator = is_inactive and
-      options.icon_separator_inactive() or
-      options.icon_separator_active()
+    --- The name of the buffer
+    --- @type bufferline.render.group
+    local name = {hl = clickable .. buffer_hl, text = buffer_name}
```

Comment:
Rather than do `local foo, fooPrefix` can then _later_ convert it to `bufferline.render.group` (i.e. `{hl = fooPrefix, text = foo}`), it made the render code a little easier to parse just to declare each component of the bufferline as a `bufferline.render.group` to begin with.